src/blockdata/transaction.rs

```diff
@@ -689,6 +707,21 @@ impl SigHashType {
          }
      }
 
+     /// Read a 4-byte uint32 as a standard sighash type, returning an error if the type
+     /// is non standard.
+     pub fn from_u32_standard(n: u32) -> Result<SigHashType, ParseSigHashTypeError> {
+         match n & 0x9f {
```

Comment:
The decoding logic is essentially duplicated. Maybe `from_u32` can be implemented by building on `from_u32_standard` (e.g. `match` or `unwrap_or_else`)?

Reply:
Right

Reply:
Actually, would it be *that* better? I would still need to `match` to catch the ACP mask

Reply:
Idk, it's marginal. Leave it if it's getting too ugly. Afaik it's unlikely to get extended.